app/assets/javascripts/utils/training_utils.js

```diff
@@ -0,0 +1,12 @@
+// utils/domUtils.js
+export const extractCategoriesFromHtml = () => {
+  const categoriesFromHtml = Array.from(document.querySelectorAll('.training__categories > li')).map((li) => {
+    const nameElement = li.querySelector('.training__category__header h1');
+    const descriptionElement = li.querySelector('.training__category__header p');
+
+    const name = nameElement ? nameElement.innerText.trim() : '';
+    const description = descriptionElement ? descriptionElement.innerText.trim() : '';
+    return { name, description };
+  });
+  return categoriesFromHtml;
+};
```

Comment:
is there not a better way to do this? Maybe pass along the list of categories to `#react_root`?

Reply:
in case you missed this @omChauhanDev